csharp/src/Ice/VectoredBufferExtensions.cs

```diff
@@ -184,5 +152,28 @@ internal static ArraySegment<byte> AsArraySegment(this IList<ArraySegment<byte>>
             }
             return dst;
         }
+
+        internal static List<ArraySegment<byte>> Slice(this IEnumerable<ArraySegment<byte>> src, int start)
+        {
+            Debug.Assert(start >= 0);
+            var result = new List<ArraySegment<byte>>();
+            foreach (ArraySegment<byte> segment in src)
+            {
+                if (start == 0)
+                {
+                    result.Add(segment);
+                }
+                else if (segment.Count > start)
+                {
+                    result.Add(segment.Slice(start));
+                    start = 0;
+                }
+                else
+                {
+                    start -= segment.Count; // and we skip this segment
+                }
+            }
```

Comment:
Should this also throw IndexOutOfRangeException if start is out of range to be consistent with other C# Slice method?